ccvm_simulators/solvers/mf_solver.py

```diff
@@ -398,6 +398,65 @@ def _optics_machine_energy_callable(
 
         return _optics_machine_energy_callable
 
+    def _optics_machine_time(self, machine_parameters: dict = None):
+        """The wrapper function of calculating the average time spent by the
+            solver on a single instance, as if the solving process was to be performed on
+            an optical MF-CCVM machine.
+
+        Args:
+            machine_parameters (dict, optional): Parameters of the optical MF-CCVM
+                machine. Defaults to None.
+
+        Raises:
+            ValueError: when the given machine parameters are not valid.
+            ValueError: when the given dataframe does not contain the required columns.
+
+        Returns:
+            Callable: A callable function that takes in a dataframe and problem size and
+                returns the average average time spent by the solver on a single instance.
+        """
+
+        if machine_parameters is None:
+            machine_parameters = self._default_optics_machine_parameters
+        else:
+            self._is_valid_optics_machine_parameters(machine_parameters)
+
+        def _optics_machine_time_callable(dataframe: DataFrame, problem_size: int):
+            """Calculate the average average time spent by the solver on a single instance,
+                simulating on a MF-CCVM machine.
+
+            Args:
+                dataframe (DataFrame): The necessary data to calculate the average
+                    time.
+                problem_size (int): The size of the problem.
+
+            Raises:
+                ValueError: when the given dataframe does not contain the required
+                    columns.
+
+            Returns:
+                float: The average average time spent by the solver on a single instance.
+            """
+            self._validate_machine_time_dataframe_columns(dataframe)
+
+            # TODO before merge: Ask Farhad if using mean was correct here
+            iterations = np.mean(dataframe["iterations"].values)
```

Comment:
@FarhadK-1QBit I assume using `np.mean()` here was intended, like Estella's PRs did, but I just want to confirm with you!

Reply:
That's right, Katie!